api/views.py

```diff
@@ -127,3 +132,79 @@ def post(self, request, *args, **kwargs):
         data = serializer.data
 
         return Response(data, status=status.HTTP_201_CREATED)
+
+    @staticmethod
+    def _get_account_from_card_id(card_uuid) -> SociBankAccount:
+        soci_bank_account = get_object_or_404(queryset=SociBankAccount.objects.all(), card_uuid=card_uuid)
+
+        return soci_bank_account
+
+
+class SensorMeasurementView(generics.CreateAPIView, generics.ListAPIView):
+    serializer_class = SensorMeasurementSerializer
+
+    def get_permissions(self):
+        if self.request.method == "POST":
+            return [SensorTokenPermission()]
+
+        return []
+
+    def get_queryset(self):
+        # We don't really bother to validate this input. A bad input will just
+        # return 0 results.
+        type = self.request.query_params.get('type', MEASUREMENT_TYPE_TEMPERATURE)
+        return SensorMeasurement.objects.filter(
+            type=type,
+            created_at__gte=timezone.now() - timezone.timedelta(days=1)
+        )
+
+    @swagger_auto_schema(
+        request_body=serializer_class(many=True),
+        operation_summary="Create measurements",
+        responses={
+            "201": serializer_class(many=True),
+            "400": ": Illegal input.",
+            "403": ": You are not authorized to create measurements."
+        },
+    )
+    def post(self, request: Request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data, many=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        # Try to clear out old entries.
+        try:
+            SensorMeasurementView._remove_old_measurement_instances()
+        except DatabaseError:
+            # Should we do something?
+            pass
```

Comment:
How can this method fail? I cannot immediately see why a filter.delete would result in this error.

Reply:
Anything related to IO can fail unexpectedly. The `DatabaseError` is a generic catch-all for Django database errors. The reason I have this here is that the clean-up of old measurement instances is not essential for the view to complete successfully. Hence some error occurring should not fail the view.

If we had a general async task processor, this code would run there instead, but for now, I find this to be the best solution.